src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
                 },
                 _joinableFactory);
 
+            _solutionLoadedEvent = new AsyncManualResetEvent(initialState: true);
```

Comment:
OK, if it doesn't matter, then fine. I wouldn't worry about the perf difference unless this was in a tight loop.